dataflow/snippets/src/test/java/com/example/dataflow/ApacheIcebergIT.java

```diff
@@ -0,0 +1,176 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dataflow;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.iceberg.data.IcebergGenerics;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.data.parquet.GenericParquetWriter;
+import org.apache.iceberg.hadoop.HadoopInputFile;
+import org.apache.iceberg.hadoop.HadoopOutputFile;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.parquet.Parquet;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.NestedField;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ApacheIcebergIT {
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private String location;
+
+  private Configuration hadoopConf = new Configuration();
+  private Catalog catalog;
+  private static final TableIdentifier tableId = TableIdentifier.of("db", "table1");
+  private Table table;
+
+  private static final String ouputFileName = "output-00000-of-00001.txt";
+
+  private void createIcebergTable(Catalog catalog, TableIdentifier tableId) throws IOException {
+    var schema = new Schema(
+        NestedField.required(1, "id", Types.LongType.get()),
+        NestedField.optional(2, "name", Types.StringType.get()));
+
+    table = catalog.createTable(tableId, schema);
+  }
+
+  private void writeRecords()
+      throws IOException {
+    GenericRecord record = GenericRecord.create(table.schema());
+    ImmutableList<Record> records =
+        ImmutableList.of(
+            record.copy(ImmutableMap.of("id", 0L, "name", "Person-0")),
+            record.copy(ImmutableMap.of("id", 1L, "name", "Person-1")),
+            record.copy(ImmutableMap.of("id", 2L, "name", "Person-2")));
```

Comment:
Is it really necessary to write 3 records? We should be looking into minimizing execution time of IT tests.
Testing the read code sample should be OK with one record.

Reply:
Fixed